python/tests/test_writer.py

```diff
@@ -304,6 +304,62 @@ def test_write_iterator(
     assert DeltaTable(tmp_path).to_pyarrow_table() == sample_data
 
 
+def test_write_dataset(
+    tmp_path: pathlib.Path, existing_table: DeltaTable, sample_data: pa.Table
+):
+    dataset = existing_table.to_pyarrow_dataset()
+
+    write_deltalake(tmp_path, dataset, mode="overwrite")
```

Comment:
This is possibly a bit convoluted, but you could instead do something like:

```suggestion
@pytest.parametrize("large_dtypes", [True, False])
@pytest.parametrize("constructor", [
    lambda table: table.to_pyarrow_dataset(),
    lambda table: table.to_pyarrow_table(),
    lambda table: table.to_pyarrow_table().to_batches()[0]
])
def test_write_dataset(
    tmp_path: pathlib.Path, existing_table: DeltaTable, sample_data: pa.Table,
    large_dtypes: bool, constructor
):
    dataset = constructor(existing_table)

    write_deltalake(tmp_path, dataset, mode="overwrite", large_dtypes=large_dtypes)
```